plugins/prow-agent/scripts/extract_metrics.py

```
@@ -0,0 +1,267 @@
+        cache_creation_tokens += mu.get("cacheCreationInputTokens", 0)
+
+    total_input = input_tokens + cache_read_tokens + cache_creation_tokens
+    cache_hit_rate = (cache_read_tokens / total_input * 100) if total_input > 0 else 0
+
+    # --- Tool usage (from assistant messages) ---
+    tool_counts = Counter()
+    skills_invoked = []
+    files_written = []
+    num_thinking_blocks = 0
+
+    for l in lines:
```

Ruff (0.15.15)

[error] Ambiguous variable name: `l`

(E741)